src/sessions/base-session-handler.ts

```diff
@@ -0,0 +1,109 @@
+import { PureCloudWebrtcSdk } from '../client';
+import { log } from '../logging';
+import { LogLevels, SessionTypes, SdkErrorTypes } from '../types/enums';
+import StatsGatherer from 'webrtc-stats-gatherer';
+import { SessionManager } from './session-manager';
+import { IPendingSession, ISessionInfo, IAcceptPendingSessionRequest, IStartSessionParams } from '../types/interfaces';
+import { checkHasTransceiverFunctionality } from '../media-utils';
+import { throwSdkError } from '../utils';
+
+export default abstract class BaseSessionHandler {
+  constructor (protected sdk: PureCloudWebrtcSdk, protected sessionManager: SessionManager) {}
+
+  abstract getSessionType (): SessionTypes;
+
+  protected log (level: LogLevels, message: any, details?: any): void {
+    log.call(this.sdk, level, message, details);
+  }
+
+  abstract shouldHandleSession ({ sessionInfo, sessionType }: { sessionInfo?: ISessionInfo, sessionType?: SessionTypes }): boolean;
+
+  async startSession (sessionStartParams: IStartSessionParams): Promise<any> {
+    throwSdkError.call(this.sdk, SdkErrorTypes.not_supported, `sessionType ${sessionStartParams.sessionType} can only be started using the purecloud api`, { sessionStartParams });
+  }
+
+  handlePropose (pendingSession: IPendingSession) {
+    this.sdk.emit('pendingSession', pendingSession);
+
+    setTimeout(() => {
+      this.sessionManager.removePendingSession(pendingSession.id);
+    }, 1000);
+  }
+
+  acceptPendingSession (session: IPendingSession, params?: IAcceptPendingSessionRequest) {
+    this.sessionManager.webrtcSessions.acceptRtcSession(session.id);
+  }
+
+  async handleSessionInit (session: any) {
+    try {
+      this.log(LogLevels.info, 'onSession', session);
+    } catch (e) {
+      // don't let log errors ruin a session
+    }
+
+    session.id = session.sid;
```

Comment:
session.id seems redundant if we're just using `sid`

Reply:
A lot of this is just to expose a "standard" variable name since we emit this to the customer.